neuron.schema.json

```diff
@@ -0,0 +1,148 @@
+                            "python": {"type": "string", "minlength": 1},
+                            "cpp": {"type": "string", "minlength": 1}
+                        },
+                        "additionalProperties": {"type": "string", "minlength": 1}
+                    }
+                },
+                "required": ["name", "derivation"],
+                "additionalProperties": false
+            }
+        },
+        "additionalParameters": {
+            "description": "The names and types of additional parameters for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    }
+                },
+                "required": ["name", "type"],
+                "additionalProperties": false
+            }
+        },
+        "additionalVariables": {
+            "description": "The names and types of additional variables for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    },
+                    "initial": {
+                        "type": "number"
+                    }
+                },
+                "required": ["name", "type", "initial"],
+                "additionalProperties": false
+            }
+        },
+        "simulationCode": {
+            "description": "The code for the simulation function of the neuron",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "thresholdConditionCode": {
+            "description": "The code describing a threshold condition",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "resetCode": {
+            "description": "The code for the reset function of the neuron",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "supportCode": {
+            "description": "Any additional support code for the neuron",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "isAutoRefactoryRequired": {
+            "description": "Whether the model requires auto-refactory logic to be generated",
+            "type": "boolean",
+            "default": false
+        },
+        "customBody": {
```

Comment:
I suppose some sort of catch-all? But what is the use case?

Reply:
again, a variable in the Python class - not really sure of its function though

Reply:
it is an option in Python but I don't think it should be (the Python interface is still a little rough around the edges) and definitely shouldn't be exposed here

Reply:
sure, will be removed.